src/consumption.cpp

```diff
@@ -351,7 +351,12 @@ std::pair<nutrients, nutrients> Character::compute_nutrient_range(
         tally_max -= byproduct_nutr;
     }
 
-    int charges = comest.count();
+    int charges = 0;
+    if( rec.makes_amount() == 0 ) {
+        charges = comest.charges;
+    } else {
+        charges = rec.makes_amount();
+    }
```

Comment:
```suggestion
    int charges = rec.makes_amount();
```

`makes_amount` can't be 0, so this should be enough. For what it's worth it's also what I do in #60885 to fix that, though I didn't even realize it was broken before.